packages/docusaurus-plugin-content-blog/src/authors.ts

```diff
@@ -196,5 +202,33 @@ Don't mix 'authors' with other existing 'author_*' front matter. Choose one or t
     return [authorLegacy];
   }
 
+  const inlineAuthors = updatedAuthors.filter((author) => author.inline);
+
+  const duplicateList = updatedAuthors.filter(
+    (author, index, self) =>
+      index !== self.findIndex((t) => t.name === author.name),
+  );
+
+  // TODO need title check otherwise reports weird cases
+  if (inlineAuthors.length > 0 && params.frontMatter.title) {
+    logger.warn(
+      `Inline authors found in blog [${
+        params.frontMatter.title
+      }] ${inlineAuthors.map((author) => author.name).join(', ')}`,
+    );
+  }
```

Comment:
We need an option `onInlineAuthors`, it must be configurable because most blogs use inline authors atm.

>   // TODO need title check otherwise reports weird cases


What are those cases exactly? We need to handle those and cover with unit test edge cases

Overall the code should look quite similar to the tags code:

```tsx
export function reportInlineTags({
  tags,
  source,
  options,
}: {
  tags: TagMetadata[];
  source: string;
  options: TagsPluginOptions;
}): void {
  if (options.onInlineTags === 'ignore') {
    return;
  }
  const inlineTags = tags.filter((tag) => tag.inline);
  if (inlineTags.length > 0) {
    const uniqueUnknownTags = [...new Set(inlineTags.map((tag) => tag.label))];
    const tagListString = uniqueUnknownTags.join(', ');
    logger.report(options.onInlineTags)(
      `Tags [${tagListString}] used in ${source} are not defined in ${
        options.tags ?? 'tags.yml'
      }`,
    );
  }
}
```

I'd prefer if the side effect (logging) wasn't performed inside `getAuthors` but as a separate standalone dedicated method, tested independently. 

```tsx
  const authors = getBlogPostAuthors({authorsMap, frontMatter, baseUrl});
  reportAuthorProblems(paramsYouNeed)
```

If logging requires more info about the blog (like the title/source/whatever) you can inject more as params.